src/sagas/vendor.js

```diff
@@ -0,0 +1,177 @@
+import { all, call, put, takeLatest, select } from 'redux-saga/effects';
+import request from 'utils/request';
+import helper from 'utils/saga';
+import { getVendor } from 'selectors/vendor';
+import { actions, types } from 'reducers/vendor';
+import { initialState, actions as vendorsActions } from 'reducers/vendors';
+
+function* requestVendorWorker({ vendorId }) {
+  try {
+    const vendor = yield select(getVendor);
+
+    if (vendor.length === 0 || vendor.vendorId !== vendorId) {
+      const endpoint = {
+        url: `/vendor/${vendorId}`,
+        method: 'GET'
+      };
+      const result = yield call(request.execute, { endpoint });
+
+      if (result.success) {
+        const {
+          response: { data }
+        } = result;
+
+        yield put(actions.requestVendorSuccess(data));
+      } else if (result.error) {
+        throw result.error;
+      } else {
+        throw new Error('Failed to get vendor!');
```

Comment:
The case where `vendor.vendorId === vendorId` probably shouldn't produce an error, but should be handled silently, since the data requested is already there... or maybe it should re-fetch it anyway, either way a little more thought is needed here.